gapidapk/deviceinfo.go

```diff
@@ -172,8 +174,18 @@ func preparePerfettoProducerLauncherFromApk(ctx context.Context, d adb.Device) (
 		return "", cleanup.Invoke(ctx), log.Errf(ctx, err, "Failed to unzip %v from %v", perfettoProducerLauncher, packageName)
 	}
 
-	// Finally, make sure the binary is executable
+	// Make sure the binary is executable
 	d.Shell("chmod", "a+x", launcherPath).Call(ctx)
+
+	// Finally, if there's an existing instance running, kill it.
+	if pid, err := d.Shell("cat", perfettoProducerLauncherPidFile).Call(ctx); err == nil {
+		if _, err := strconv.Atoi(pid); err == nil {
+			if res, err := d.Shell("perfetto", "--query").Call(ctx); err == nil && (strings.Contains(res, "gpu.counters") || strings.Contains(res, "gpu.renderstages")) {
+				d.Shell("kill -2", pid).Call(ctx)
+			}
+		}
+	}
+
```

Comment:
Instead of this, we should read the file in the producer and just kill the pid in the file, if found.

Reply:
I add check to only kill process when producers already exist. It's easier for us to do it here I think

Reply:
i don't think we should do that check, I don't see any upside to the check.